tensorboard/plugins/hparams/backend_context_test.py

```diff
@@ -251,22 +254,33 @@ def test_experiment_with_session_tags(self):
                 min_value: 100.0
                 max_value: 300.0
               }
-            },
+              differs: true
+            }
+            hparam_infos: {
+              name: 'eval.timeout'
+              type: DATA_TYPE_BOOL
+              domain_discrete: {
+                values: [{bool_value: false}]
+              }
+              differs: false
```

Comment:
Would it be relatively simple to add the following cases, which I don't think are covered:

1. type: DATA_TYPE_BOOL and differs: true
2. type: DATA_TYPE_FLOAT64 and differs: false
3. type: DATA_TYPE_STRING and differs: false

Is it reasonably possible to cover all 6 cases in a single devoted test?

Reply:
Yes, made this test more devoted 🐢

Reply:
To be "more devoted" I would have preferred the original "test_experiment_with_session_tags" to be left as-is as much as possible but then add a "test_experiment_with_session_tags_differs_field" to be the "devoted" test. As you've written it I feel we have one test trying to cover too many behaviors.

The bonus is you could give the hparams names like "float_hparam_differs_true", "bool_hparam_differs_false", etc.

Reply:
Done.